app/models/search/work_creator_indexer.rb

```diff
@@ -12,14 +12,13 @@ def self.mapping
   def routing_info(id)
     {
       "_index" => index_name,
-      "_type" => document_type,
       "_id" => document_id(id),
       "routing" => parent_id(id, nil)
     }
   end
 
   def document_id(id)
-    "#{id}-creator"
+    id.to_s.ends_with?("creator") ? id : "#{id}-creator"
```

Comment:
Maybe it'd be good to add a [find_elasticsearch_ids](https://github.com/otwcode/otwarchive/blob/b63fce67a3975ec0614bb535c75d76c8dafbda71/app/models/search/bookmarkable_indexer.rb#L15-L18) method here, instead? If there's already something with an id of `9-creator-creator-creator` knocking around in the system, this would let it slide, instead of fixing it and making it `9-creator`.

Reply:
I'll try out a find_elasticsearch_ids and see how it goes 👍 . Before my change, the [tests on this branch were failing](https://github.com/otwcode/otwarchive/runs/2375780750?check_suite_focus=true#step:13:4522) with the new and exciting `Validation Failed: 1: id [57-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator-creator] is too long, must be no longer than 512 bytes but was: 514;2:` so it should be easy to see if it's working.

Reply:
Hmm, I believe that error only shows up if there are persistent indexing failures unrelated to the `-creator` suffix issue (since with this bug, each `-creator` suffix added represents one pass through the failures queue, so that particular ID represents the end of a long sequence of indexing errors). Did you track down the underlying indexing error? If not, it may still be failing silently.

Reply:
Actually, you're right, the error that was making it do that became apparent when I tweaked the id: there was a nested array in the query that ES didn't like which I fixed with the [other change in this commit](https://github.com/otwcode/otwarchive/pull/3570/commits/899b416a98c684fb253111d02af98f40b16686b7#diff-e018acad1b34c8885b0293476d3257a17cf82055a34e1e69a1374986b4a3ac56R21)  (I was able to troubleshoot locally so both changes ended up in the same commit)

Reply:
Was that really the cause? Does it work if you disable the `-creator` fix? Because that looks like it'd only make the query fail, not the indexing. (And it looks like it's for a different index, too.)

Reply:
I just looked at the test runs for more of the commits, and it looks like it may have been fixed in the commit before the one you linked: https://github.com/otwcode/otwarchive/pull/3570/commits/b6ffa99f8f496bce148b4cdc20ea6e2f8b2abac1. Not sure why that would be marked as an error instead of a warning, but it's related to indexing and it does involve the right index, so I think it's a more likely candidate than an issue with the pseud query.

(I double-checked locally, and it looks like with `_type` included in the routing info, the indexing error is `{"type"=>"illegal_argument_exception", "reason"=>"can't merge a non object mapping [creator_join] with an object mapping"}`. I'm guessing that maybe specifying a type caused something to go wrong with the lookup for the mapping? IDK.)

Reply:
Well, I've now also removed the find_elasticsearch_ids method you'd suggested the other day and the tests are still passing without the `creator-creator-creator` ids being generated, so whatever it was, it appears to be fixed now without any unnecessary changes to this file. 🎉

Reply:
The `find_elasticsearch_ids` method is still a good bug fix to have (either here or in a different PR). The `creator-creator-creator` issue is a bug in the error-handling system; removing the deprecated `_type` prevents the error that caused the error-handling system to run (and thus prevents the bug in the error-handling system from showing up in tests), but the bug still does need to be addressed.